FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc

```diff
@@ -272,347 +288,349 @@ const SeedingNode<TrackingLayer>* TrajectorySeedProducer::insertHit(
+            {
+                break; //continues with the next seeding algorithm
+            }
+
+            const AlgebraicSymMatrix55& m = initialTSOS.localError().matrix();
+            int dim = 5; /// should check if corresponds to m
+            float localErrors[15];
+            int k = 0;
+            for (int i=0; i<dim; ++i)
+            {
+                for (int j=0; j<=i; ++j)
+                {
+                    localErrors[k++] = m(i,j);
+                }
+            }
+            int surfaceSide = static_cast<int>(initialTSOS.surfaceSide());
+            initialState = PTrajectoryStateOnDet( initialTSOS.localParameters(),initialTSOS.globalMomentum().perp(),localErrors, recHits.back().geographicalId().rawId(), surfaceSide);
+            output->push_back(TrajectorySeed(initialState, recHits, PropagationDirection::alongMomentum));
+
+        }
+    } //end loop over simtracks
     
 
-	e.put(output,outputSeedCollectionName);
+    e.put(output);
 }
 
 
+
 bool
-TrajectorySeedProducer::compatibleWithBeamAxis(
-        const GlobalPoint& gpos1, 
+TrajectorySeedProducer::compatibleWithBeamSpot(
+        const GlobalPoint& gpos1,
         const GlobalPoint& gpos2,
         double error,
         bool forward
-    ) const 
+    ) const
 {
 
-    const double x0 = beamspotPosition.X();
-    const double y0 = beamspotPosition.Y();
-    const double z0 = beamspotPosition.Z();
-    
-    if ( !seedCleaning ) 
-    {
-        return true;
-    }
-
     // The hits 1 and 2 positions, in HepLorentzVector's
     XYZTLorentzVector thePos1(gpos1.x(),gpos1.y(),gpos1.z(),0.);
     XYZTLorentzVector thePos2(gpos2.x(),gpos2.y(),gpos2.z(),0.);
-    
-    // Create new particles that pass through the second hit with pT = ptMin 
-    // and charge = +/-1
 
-    // The momentum direction is by default joining the two hits 
+    // Create new particles that pass through the second hit with pT = ptMin
+    // and charge = +/-1
+    // The momentum direction is by default joining the two hits
     XYZTLorentzVector theMom2 = (thePos2-thePos1);
 
     // The corresponding RawParticle, with an (irrelevant) electric charge
     // (The charge is determined in the next step)
     ParticlePropagator myPart(theMom2,thePos2,1.,magneticFieldMap);
 
-    /// Check that the seed is compatible with a track coming from within
-    /// a cylinder of radius originRadius, with a decent pT, and propagate
-    /// to the distance of closest approach, for the appropriate charge
-    bool intersect = myPart.propagateToBeamCylinder(thePos1,originRadius*1.);
-    if ( !intersect ) 
+    /*
+    propagateToBeamCylinder does the following
+       - check there exists a track through the 2 hits and through a
+    cylinder with radius "originRadius" centered around the CMS axis
+       - if such tracks exists, pick the one with maximum pt
+       - track vertex z coordinate is z coordinate of closest approach of
+    track to (x,y) = (0,0)
+    */
+    if (originRadius>0)
```

Comment:
we agreed that setting a selection to <0 deactivates the specific cut; do not know if this may lead to meaningless behavior if this cut is deactivated... Maybe a warning can be logged?